pkg/gateways/http/transfer/check_credential.go

```diff
@@ -0,0 +1,11 @@
+package transfer
+
+import "strings"
+
+func CheckCredential(header string) (string, error) {
+	headerSplit := strings.Split(header, " ")
+	if "Basic" != headerSplit[0] {
+		return "", ErrInvalidCredential
+	}
+	return headerSplit[1], nil
+}
```

Comment:
nice job, that's the idea...although CheckCredential is not a nice name since what you're doing is just checking the authorization header type. besides, by reading the error returned is not clear that the problem is on Authorization header type